src/main/java/ui/UI.java

```diff
@@ -217,6 +221,16 @@ private void initPreApplicationState() {
 
         uiManager = new UIManager(this);
         status = new HTStatusBar(this);
+
+        updateProgressWindow = new UpdateProgressWindow();
+
+        if (!TestController.isTestMode()) {
+            updateManager = new UpdateManager(updateProgressWindow);
+        } else {
+            updateManager = new UpdateManagerStub();
```

Comment:
TestController. There are a few methods to conditionally create similar things already in there.

Also you may be able to use mockito to stub it, then you don't need a new class.